src/vm/costs/mod.rs

```diff
@@ -842,45 +906,53 @@ impl CostTracker for LimitedCostTracker {
         cost_function: ClarityCostFunction,
         input: &[u64],
     ) -> std::result::Result<ExecutionCost, CostErrors> {
-        if self.free {
-            return Ok(ExecutionCost::zero());
-        }
-        if cost_function == ClarityCostFunction::Unimplemented {
-            info!("Used unimplemented cost function");
-            return Ok(ExecutionCost::zero());
-        }
-
-        let cost_function_ref = self
-            .cost_function_references
-            .get(&cost_function)
-            .ok_or(CostErrors::CostComputationFailed(format!(
-                "CostFunction not defined: {}",
-                &cost_function
-            )))?
-            .clone();
+        match self {
+            Self::Free => {
+                // tracker is free, return zero!
+                return Ok(ExecutionCost::zero());
+            }
+            Self::Limited(ref mut data) => {
+                if cost_function == ClarityCostFunction::Unimplemented {
```

Comment:
Should this branch be a runtime panic?

Reply:
Not yet -- the 2.10 branch uses `Unimplemented` for cost functions that haven't been specified yet. There's an open issue for resolving.